doc/languages-frameworks/go.section.md

```diff
@@ -62,6 +62,82 @@ The following is an example expression using `buildGoModule`:
 }
 ```
 
+### Obtaining and overriding `vendorHash` for `buildGoModule` {#buildGoModule-vendorHash}
+
+`nix-prefetch` can be used to obtain the actual hash. The following command gets the value of `vendorHash` for package `pet`:
+
+```sh
+cd path/to/nixpkgs
+nix-prefetch -E "{ sha256 }: ((import ./. { }).my-package.overrideAttrs (_: { vendorHash = sha256; })).go-modules"
+```
+
+To obtain the hash without external tools, set `vendorHash = lib.fakeSha256;` and run the build. ([more details here](#sec-source-hashes)).
+
+`vendorHash` can be overridden with `overrideAttrs`. Override the above example like this:
+
+```nix
+{
+  pet_0_4_0 = pet.overrideAttrs (
+    finalAttrs: previousAttrs: {
+      version = "0.4.0";
+
+      src = fetchFromGitHub {
+        inherit (previousAttrs.src) owner repo;
+        rev = "v${finalAttrs.version}";
+        hash = "sha256-gVTpzmXekQxGMucDKskGi+e+34nJwwsXwvQTjRO6Gdg=";
+      };
+
+      vendorHash = "sha256-dUvp7FEW09V0xMuhewPGw3TuAic/sD7xyXEYviZ2Ivs=";
+    }
+  );
+}
+```
+
+### Overriding `goModules` (#buildGoModule-goModules-override)
+
+As `goModules` would become `""` when the final state of attribute `vendorHash` becomes `null`, overriding `goModules` directly with `goModules.overrideAttrs` may cause error when subsequent overriding specifies `venhorHash = null`.
```

Comment:
A humble suggestion:

```suggestion
When the final state of attribute `vendorHash` becomes `null`, `goModules` becomes `""`. Hence, overriding `goModules` directly with `goModules.overrideAttrs` may cause error when subsequent overriding specifies `venhorHash = null`.
```

Another suggestion:

```suggestion
In the past, overriding `${pkg}.goModules` was the only way to override them (see [#86349](https://github.com/NixOS/nixpkgs/issues/86349)). Now, it is possible to directly override them by overriding the main derivation, and overriding `${pkg}.goModules` is unsupported. This is because when the final state of attribute `vendorHash` becomes `null`, `goModules` becomes `""`.
```

It is better to write the main point of the sentence first IMO.

Reply:
Thank you very much for rephrasing it!

> In the past, overriding `${pkg}.goModules` was the only way to override them

It was also impossible to override `goModules` since it was not even exposed as an overridable attribute (overriding `passthru.goModules` doesn't affect the primary derivation). The only workaround is a hack of the packages dependent `buildGoModule` through the `<pkg>.override` interface (https://github.com/NixOS/nixpkgs/issues/86349#issuecomment-624489806).

With this PR applied, we could override the `venhorHash` and `modRoot` attributes from the primary derivation. As for other overriding specific to `goModules`,  I would like to inform the reader to override `passthru.overrideModAttrs` instead.